### PR TITLE
Chore[New exp]: HTTP Modify Headers experiment for K8s 

### PR DESCRIPTION
**Pull request by avaakash**
**What this PR does / why we need it**:
This PR adds the HTTP Modify Header experiment which can modify the request/response headers. It can be used to 
- Add headers
- Modify Header values
- Remove Headers (can only empty the header value, not specifically able to remove the header)

The current implementation is to use nsenter to run two commands
The first is to start the toxiproxy proxy server and then add the toxic for the provided target host and port.
Next is to add a ruleset in IPtables to direct the request coming for the target port to the proxy port. This will remove the requirement to change the ports manually.

**Special notes for your reviewer**:
Related PRs
- https://github.com/litmuschaos/litmus/pull/3684
- https://github.com/litmuschaos/chaos-charts/pull/558

#### Experiment Pod Logs
```
time="2022-07-13T10:11:57Z" level=info msg="Experiment Name: pod-http-modify-header"
time="2022-07-13T10:11:57Z" level=info msg="[PreReq]: Getting the ENV for the pod-http-modify-header experiment"
time="2022-07-13T10:11:59Z" level=info msg="[PreReq]: Updating the chaos result of pod-http-modify-header experiment (SOT)"
time="2022-07-13T10:12:01Z" level=info msg="The application information is as follows\n" Chaos Duration=60 Container Runtime=containerd Namespace=default Label="app=nginx"
time="2022-07-13T10:12:01Z" level=info msg="[Status]: Verify that the AUT (Application Under Test) is running (pre-chaos)"
time="2022-07-13T10:12:01Z" level=info msg="[Status]: Checking whether application containers are in ready state"
time="2022-07-13T10:12:01Z" level=info msg="[Status]: The Container status are as follows" container=nginx Pod=nginx-deployment-66b6c48dd5-9zkcs Readiness=true
time="2022-07-13T10:12:03Z" level=info msg="[Status]: Checking whether application pods are in running state"
time="2022-07-13T10:12:03Z" level=info msg="[Status]: The status of Pods are as follows" Pod=nginx-deployment-66b6c48dd5-9zkcs Status=Running
time="2022-07-13T10:12:05Z" level=info msg="[Info]: The chaos tunables are:" Listen Port=2002 Sequence=parallel PodsAffectedPerc=100 Target Port=80
time="2022-07-13T10:12:05Z" level=info msg="[Chaos]:Number of pods targeted: 1"
time="2022-07-13T10:12:05Z" level=info msg="[Info]: Target pods list for chaos, [nginx-deployment-66b6c48dd5-9zkcs]"
time="2022-07-13T10:12:05Z" level=info msg="[Info]: Details of application under chaos injection" PodName=nginx-deployment-66b6c48dd5-9zkcs NodeName=gke-hce-test-plan-re-default-pool-f2bec7be-p45t ContainerName=nginx
time="2022-07-13T10:12:05Z" level=info msg="[Status]: Checking the status of the helper pods"
time="2022-07-13T10:12:07Z" level=info msg="pod-http-modify-header-helper-dyywtf helper pod is in Running state"
time="2022-07-13T10:12:09Z" level=info msg="[Wait]: waiting till the completion of the helper pod"
time="2022-07-13T10:12:09Z" level=info msg="helper pod status: Running"
time="2022-07-13T10:12:10Z" level=info msg="helper pod status: Running"
time="2022-07-13T10:13:17Z" level=info msg="helper pod status: Running"
time="2022-07-13T10:13:18Z" level=info msg="helper pod status: Running"
time="2022-07-13T10:13:18Z" level=info msg="[Status]: The running status of Pods are as follows" Status=Running Pod=pod-http-modify-header-helper-dyywtf
time="2022-07-13T10:13:19Z" level=info msg="[Cleanup]: Deleting all the helper pod"
time="2022-07-13T10:13:21Z" level=info msg="[Confirmation]: pod-http-modify-header chaos has been injected successfully"
time="2022-07-13T10:13:21Z" level=info msg="[Status]: Verify that the AUT (Application Under Test) is running (post-chaos)"
time="2022-07-13T10:13:21Z" level=info msg="[Status]: Checking whether application containers are in ready state"
time="2022-07-13T10:13:22Z" level=info msg="[Status]: The Container status are as follows" Pod=nginx-deployment-66b6c48dd5-9zkcs Readiness=true container=nginx
time="2022-07-13T10:13:24Z" level=info msg="[Status]: Checking whether application pods are in running state"
time="2022-07-13T10:13:24Z" level=info msg="[Status]: The status of Pods are as follows" Pod=nginx-deployment-66b6c48dd5-9zkcs Status=Running
time="2022-07-13T10:13:26Z" level=info msg="[The End]: Updating the chaos result of pod-http-modify-header experiment (EOT)"
```

#### Helper Pod Logs
```
time="2022-07-13T10:12:06Z" level=info msg="Helper Name: http-chaos"
time="2022-07-13T10:12:06Z" level=info msg="[PreReq]: Getting the ENV variables"
time="2022-07-13T10:12:06Z" level=info msg="container ID of nginx container, containerID: 924eaa83e443f187d0ef43e5821384cc0a3d839a57968777e8511058e9881f11"
time="2022-07-13T10:12:06Z" level=info msg="Container ID: 924eaa83e443f187d0ef43e5821384cc0a3d839a57968777e8511058e9881f11"
time="2022-07-13T10:12:06Z" level=info msg="[Info]: Container ID=924eaa83e443f187d0ef43e5821384cc0a3d839a57968777e8511058e9881f11 has process PID=3098"
time="2022-07-13T10:12:06Z" level=info msg="[Chaos]: Starting proxy server: /bin/bash -c (sudo nsenter -t 3098 -n toxiproxy-server -host=0.0.0.0 > /dev/null 2>&1 &) && sleep 10 && (sudo nsenter -t 3098 -n toxiproxy-cli create -l 0.0.0.0:2002 -u 0.0.0.0:80 proxy) && (sudo nsenter -t 3098 -n toxiproxy-cli toxic add -t header --downstream -a headers='{\"new\":\"test\", \"Content-Type\": \"text/text\"}' -a mode=response proxy)"
time="2022-07-13T10:12:16Z" level=info msg="[Info]: Proxy started successfully"
time="2022-07-13T10:12:16Z" level=info msg="[Chaos]: Adding IPtables ruleset: /bin/bash -c (sudo nsenter -t 3098 -n iptables -t nat -A PREROUTING -i eth0 -p tcp --dport 80 -j REDIRECT --to-port 2002)"
time="2022-07-13T10:12:16Z" level=info msg="[Info]: IP rule set added successfully"
time="2022-07-13T10:12:18Z" level=info msg="[Chaos]: Waiting for 60s"
time="2022-07-13T10:13:18Z" level=info msg="[Chaos]: Stopping the experiment"
time="2022-07-13T10:13:18Z" level=info msg="[Chaos]: Removing IPtables ruleset: /bin/bash -c sudo nsenter -t 3098 -n iptables -t nat -D PREROUTING -i eth0 -p tcp --dport 80 -j REDIRECT --to-port 2002"
time="2022-07-13T10:13:18Z" level=info msg="[Info]: IP rule set removed successfully"
time="2022-07-13T10:13:18Z" level=info msg="[Chaos]: Stopping proxy server: /bin/bash -c sudo nsenter -t 3098 -n sudo kill -9 $(ps aux | grep [t]oxiproxy | awk 'FNR==1{print $1}')"
time="2022-07-13T10:13:18Z" level=info msg="[Info]: Proxy stopped successfully"
```

### CODE REVIEW

File: chaoslib/litmus/http-chaos/lib/header/header.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+)
+
+//PodHttpModifyHeaderChaos contains the steps to prepare and inject http latency chaos
+func PodHttpModifyHeaderChaos(experimentsDetails *experimentTypes.ExperimentDetails, clients clients.ClientSets, resultDetails *types.ResultDetails, eventsDetails *types.EventDetails, chaosDetails *types.ChaosDetails) error {
+	var stream string
+	if experimentsDetails.HeaderMode == "request" {
+		stream = "upstream"
+	} else {
+		stream = "downstream"
+	}
```
Comment (neelanjan00) [inline chaoslib/litmus/http-chaos/lib/header/header.go:17]:
```suggestion
	stream := "downstream"
	if experimentsDetails.HeaderMode == "request" {
		stream = "upstream"
	} 
```

**Review by uditgaurav — APPROVED** [review]

File: chaoslib/litmus/http-chaos/lib/header/header.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package latency
```
Comment (uditgaurav) [inline chaoslib/litmus/http-chaos/lib/header/header.go:1]:
```suggestion
package header
```

File: chaoslib/litmus/http-chaos/lib/header/header.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import (
+	http_chaos "github.com/litmuschaos/litmus-go/chaoslib/litmus/http-chaos/lib"
+	clients "github.com/litmuschaos/litmus-go/pkg/clients"
+	experimentTypes "github.com/litmuschaos/litmus-go/pkg/generic/http-chaos/types"
+	"github.com/litmuschaos/litmus-go/pkg/log"
+	"github.com/litmuschaos/litmus-go/pkg/types"
+	"github.com/sirupsen/logrus"
+)
+
+//PodHttpModifyHeaderChaos contains the steps to prepare and inject http latency chaos
```
Comment (uditgaurav) [inline chaoslib/litmus/http-chaos/lib/header/header.go:12]:
```suggestion
//PodHttpModifyHeaderChaos contains the steps to prepare and inject http header chaos
```

File: experiments/generic/pod-http-modify-header/experiment/pod-http-modify-header.go [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+	"github.com/litmuschaos/litmus-go/pkg/log"
+	"github.com/litmuschaos/litmus-go/pkg/probe"
+	"github.com/litmuschaos/litmus-go/pkg/result"
+	"github.com/litmuschaos/litmus-go/pkg/status"
+	"github.com/litmuschaos/litmus-go/pkg/types"
+	"github.com/litmuschaos/litmus-go/pkg/utils/common"
+	"github.com/sirupsen/logrus"
+)
+
+// PodHttpModifyHeader inject the pod-http-latency chaos
```
Comment (uditgaurav) [inline experiments/generic/pod-http-modify-header/experiment/pod-http-modify-header.go:21]:
```suggestion
// PodHttpModifyHeader inject the pod-http-modify-header chaos
```

**Review by uditgaurav — CHANGES_REQUESTED** [review]

**Review by uditgaurav — APPROVED** [review]

**Review by Jonsy13 — APPROVED** [review]
Looks good! 🚀

**Review by neelanjan00 — APPROVED** [review]

**Pull request merged**